src/authorize.js

```diff
@@ -36,6 +36,22 @@ function extractParams(url) {
   }, {});
 }
 
+function buildOAuthURL (authURL, redirectUri, scope, clientId) {
+  const hashPos = redirectUri.indexOf('#');
+  let url = authURL;
+
+  url += authURL.indexOf('?') > 0 ? '&' : '?';
+  url += 'redirect_uri=' + encodeURIComponent(redirectUri.replace(/#.*$/, ''));
+  url += '&scope=' + encodeURIComponent(scope);
+  url += '&client_id=' + encodeURIComponent(clientId);
+
+  if (hashPos !== - 1 && hashPos+1 !== redirectUri.length) {
+    url += '&state=' + encodeURIComponent(redirectUri.substring(hashPos+1));
+  }
+  url += '&response_type=token';
+
+  return url;
+}
```

Comment:
I can't see a test for this function in the commit.

Reply:
I only extracted this to a (private) function from `Authorize()`. Everything is already tested implicitly by the `Authorize()` tests.

In fact you can change any of the lines in this method and will result in a test failure.

Reply:
Yes, but as it's not part of the tested function anymore, this connection is not clear to anyone who's not you (which even includes future-you).

Reply:
It's a private function and you are not supposed to test private functions. I can't even call the function from the tests.

Reply:
Ok, makes sense. Thanks for explaining.